appcues/src/main/java/com/appcues/ui/AppcuesViewModel.kt

```diff
@@ -84,13 +112,40 @@ internal class AppcuesViewModel(
         }
     }
 
+    fun onPresentationComplete() {
+        uiState.value.let { state ->
+            if (state is Rendering) {
+                appcuesCoroutineScope.launch {
+                    state.presentationComplete(Success(Unit))
+                }
+            }
+        }
+    }
+
     fun onActivityChanged() {
         uiState.value.let { state ->
             // if current state IS Rendering this means that the Activity was removed
             // from an external source (ex deep link) and we should end the experience
             if (state is Rendering) {
                 appcuesCoroutineScope.launch {
-                    experienceRenderer.dismiss(renderContext, markComplete = false, destroyed = true)
+                    // this is to avoid any state machine hang waiting for this callback, if, the experience
+                    // was destroyed before presentation completed. Normally, the presentation complete callback
+                    // will have already been triggered during first render, and this will do nothing.
+                    val handled = state.presentationComplete(
+                        Failure(
+                            StepError(
+                                experience = state.experience,
+                                stepIndex = state.flatStepIndex,
+                                message = "Activity changed during render of step ${state.flatStepIndex}",
+                            )
+                        )
+                    )
```

Comment:
this is where we handle the new edge case - the Activity was removed during the error handling / retry phase.